### ISSUE TITLE
ServiceObject#parent could be a ServiceObject

### ISSUE THREAD

**Issue by kinwa91** [googleapis/nodejs-common#180]
https://github.com/googleapis/nodejs-common/blob/master/src/service-object.ts#L74

The interface declaration for `ServiceObjectConfig` specify its `parent` property to be a Service object. In the context of `nodejs-storage`. While this is true for `Bucket`, whose parent is `Storage`, he parent of `File` is `Bucket`, which is a `ServiceObject`.

https://github.com/googleapis/nodejs-storage/blob/master/src/file.js#L2266
```typescript
class File extends common.ServiceObject {
  ...
  setMetadata(metadata, options, callback) {
    ...
    this.parent.setMetadata.call(this, metadata, options, callback);
  }
```

**Comment by kinwa91** [comment]
What do y'all think about this implementation?

https://github.com/googleapis/nodejs-common/compare/master...kinwa91:fix-nested-service-obj-subclass

It essentially allows `ServiceObject#parent` to be either a `Service` or `ServiceObject` object

**Comment by kinwa91** [comment]
Closed via #212
